src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStepExecution.java

```diff
@@ -253,10 +253,20 @@ private PodTemplateCallback(PodTemplate podTemplate) {
          * Remove the template after step is done
          */
         protected void finished(StepContext context) throws Exception {
-            Cloud cloud = Jenkins.get().getCloud(cloudName);
+            Cloud cloud;
```

Comment:
Looks simpler to just call `resolveCloud`. Why duplicate the logic?

Reply:
It seemed like an error should not occur, but instead logging for whether or not the template could be removed. If an exception being thrown is not an issue it could be updated.